docs/spec/index.md

```diff
@@ -139,3 +141,120 @@ Because some events may overlap in time, it may not be possible to choose a sing
+| `date` | Date | Required | Date when exception occurs. |
+| `exception_type` | Enum | Required | `1` - The run is added to this roster for the specified date.<br />`2` - The roster will not work its regular run on this date. |
+| `service_id` | ID referencing `run_events.txt` | Conditionally Required | Part of the Run ID, which is refered to as `(service_id, run_id)`. Required if `run_id` is present. |
+| `run_id` | ID referencing `run_events.txt` | Conditionally Required | The run that's either added or removed from this roster. Required if `exception_type` is `1`. Optional if `exception_type` is `2`. If `exception_type` is `2` and `run_id` is not blank, then it must match the Run ID that the roster was scheduled to do on this date according to `rosters.txt` |
+
+### `employee_rosters.txt`
+
+Describes which employees are scheduled to which rosters on which dates.
+
+Primary Key: `(roster_id,start_date)`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `roster_id` | ID referencing `rosters.txt` | Required | |
+| `start_date` | Date | Required | |
+| `end_date` | Date | Required | |
+
+Each roster can only be assigned to one employee on each date. Employees may be scheduled to more than one roster on the same date.
+
+### `employee_run_dates.txt`
+
+Describes which employees are scheduled to which runs on which dates. If `employee_rosters.txt` is used, then describes exceptions to that schedule.
+
+Primary Key: `*`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `date` | Date | Required | |
+| `service_id` | ID referencing `run_events.txt` | Required | Part of the Run ID, which is refered to as `(service_id, run_id)`. |
+| `run_id` | ID referencing `run_events.txt` | Required | The run that's either added or removed from this employee's schedule. If `exception_type` is `2` and `run_id` is not blank, then it must match a Run ID that the employee was scheduled to do on this date according to `employee_rosters.txt`, `rosters.txt` and `roster_dates.txt`. |
+| `exception_type` | Enum | Optional | `1` (or blank) - The run is assigned to this employee on the specified date.<br />`2` - The employee will not work this run on this date. |
+
+If a feed doesn't represent rosters, it can still assign employees to runs by putting every run for every date in this file. In that case, the `exception_type` column can be omitted because every row would be adding a date, which is the default when the column is blank.
+
+Each run can only be assigned to one employee on each date. Employees may be scheduled to more than one run on the same date.
+
+### `vehicle_assignments.txt`
+
+Primary Key: `*`
```

Comment:
This key allows multiple vehicles/types to be assigned to the same block/trip. Is that intentional, as a way to allow any of the vehicles to run the trip? If so, then that should be mentioned in the docs. If not, then a primary key that would prevent that is `(date, block_id, trip_id)`.

Reply:
Hmm… I'll add a primary key. `(date, block_id, service_id)`.
Allowing multiple assignments or planning for couplings seems like a complex case.

Reply:
85f67e58a8a68ed06d1b0e1ee4ffbd55d3e294ba